provider/oauth2/tests.py

```diff
@@ -579,7 +593,7 @@ def test_clear_expired(self):
         self.assertTrue('code' in location)
 
         # verify that Grant with code exists
-        code = urlparse.parse_qs(location)['code'][0]
+        code = urlparse.parse_qs(location.split('?')[1])['code'][0]
```

Comment:
nice catch here.  though it's slightly less concise, `urlparse.urlparse(location).query` would be clearer here.